### DISCUSSION TITLE
Trying to replace dmenu_run with Alacritty + fzf, but the last step is challenging.

### DISCUSSION THREAD

**Discussion by ForeverNooob** [i3/i3#5250] [Q&A]
Hi,

I want to replace `dmenu_run` with `alacritty` + `fzf`.

I have this script (called `listexecfiles_with_fzf.sh`) which lists all executable files in my `$PATH` and then yeets it over to `fzf`: https://paste.debian.net/1259659/

So far so good. Now, I want to launch it in an alacritty window, and once I select an executable, I want the alacritty window to terminate but the executable should still remain opened and running. So... basically similar behavior to `dmenu_run`

I've managed to achieve this in the terminal by running this script (let's call this one `i3int.sh`:

```bash
#!/usr/bin/env sh

exec i3-msg "exec --no-startup-id $(~/test/listexecfiles_with_fzf.sh)"
```

And I'm calling _that_ script using something which we can call `launchalacritty.sh`

```bash
#!/usr/bin/env sh

exec alacritty --class 'launcher' --command ~/test/i3int.sh
```

All of this works as expected. Meaning: if I do `./launchalacritty.sh` in my shell, it works as expected.
But how do I set this up correctly inside my i3 config?

* If I just do: `bindsym $mod+d exec --no-startup-id alacritty --class 'launcher' --command ~/test/i3int.sh` - This doesn't work.
* If I do: `bindsym $mod+d exec ~/test/launchalacritty.sh` - This also doesn't work.
  

So what am I doing wrong here?

**Comment by okraits** [comment]
I haven't looked into this myself but @meribold does something similar:

https://www.youtube.com/watch?v=kw2mnwhptjw

Take a look at his dotfiles:

https://github.com/meribold/dotfiles

**Reply by ForeverNooob** [reply]
Thanks, but it appears that his solution can only run `.desktop` files, and relies on some Bash tricks? Whereas I want to keep my solution as POSIX as possible (so that I can switch to a different POSIX compatible shell like Zsh and still have everything working)

I did manage to get it working (see my other comment below), but I think it can use some improvement.

**Comment by ForeverNooob** [comment]
Ok, apparently the culprit was that `fzf` wasn't properly specified in my path, so I had to explicitly specify its full path.

But... I'm wondering if this can't become more elegant, because as of right now all of this looks like a weird Rube Goldberg machine of scripts. To illustrate:

* In my `~/.config/i3/config` I have the following: `bindsym $mod+d exec --no-startup-id alacritty --class 'fzf_launcher' --command sh -c '~/test/i3int.sh'`
* The contents of `~/test/i3int.sh` are: `exec i3-msg "exec --no-startup-id $(~/test/listexecfiles_with_fzf.sh)"`
* The contents of `~/test/listexecfiles_with_fzf.sh` is that [paste](https://paste.debian.net/1259659/) I linked earlier, with the only difference being that I specify the path of `fzf` explicitly.

So yeah, is there a better way to do this?

**Reply by ForeverNooob** [reply]
Update: I've managed to combine `~/test/i3int.sh` and `~/test/listexecfiles_with_fzf.sh`, so now the whole thing uses one less file: https://0x0.st/oE67.sh

...but now I've realized that it doesn't go on top of fullscreened windows. Since that is probably out of scope for this question, I've made a separate question just for that: https://github.com/i3/i3/discussions/5263
